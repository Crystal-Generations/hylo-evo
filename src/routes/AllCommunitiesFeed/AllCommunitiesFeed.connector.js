import { connect } from 'react-redux'
import { getMe } from 'store/selectors/getMe'
import getParam from 'store/selectors/getParam'
import getQueryParam from 'store/selectors/getQueryParam'
import getTopicForCurrentRoute from 'store/selectors/getTopicForCurrentRoute'
import { FETCH_POSTS } from 'store/constants'
import { mapDispatchToPropsForFeed } from 'routes/CommunityFeed'
import { fetchTopics } from './AllCommunitiesFeed.store'

export function mapStateToProps (state, props) {
  const filter = getQueryParam('t', state, props)
  const sortBy = getQueryParam('s', state, props)
  const topic = getTopicForCurrentRoute(state, props)

  return {
    filter,
    sortBy,
    topic,
    topicName: getParam('topicName', state, props),
    selectedPostId: getParam('postId', state, props),
    pending: state.pending[FETCH_POSTS],
    currentUser: getMe(state, props)
  }
}

export const mapDispatchToProps = function (dispatch, props) {
  const name = getParam('topicName', null, props)
  return {
    ...mapDispatchToPropsForFeed(dispatch, props),
    fetchTopic: () => dispatch(fetchTopics({name, first: 1}))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
