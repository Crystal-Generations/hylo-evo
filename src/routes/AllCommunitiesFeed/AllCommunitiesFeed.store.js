import { FETCH_TOPICS } from 'store/constants'

const query =
`query ($name: String, $autocomplete: String, $first: Int, $offset: Int) {
  communityTopics(topicName: $topicName, communitySlug: $communitySlug, first: $first, offset: $offset, autocomplete: $autocomplete) {
    total
    hasMore
    items {
      id
      postsTotal
      followersTotal
      topic {
        id
        name
      }
      community {
        id
      }
    }
  }
}`

export function fetchTopics (opts) {
  return {
    type: FETCH_TOPICS,
    graphql: {
      query,
      variables: {
        name: opts.name,
        autocomplete: opts.autocomplete,
        first: opts.first,
        offset: opts.offset
      }
    },
    meta: {
      extractModel: 'Topic'
    }
  }
}
