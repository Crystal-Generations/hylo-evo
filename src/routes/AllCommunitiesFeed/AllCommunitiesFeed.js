import React, { Component } from 'react'
import './AllCommunitiesFeed.scss'
import Feed from 'components/Feed'
import FeedBanner from 'components/FeedBanner'
import TopicFeedHeader from 'components/TopicFeedHeader'
import { get, pick } from 'lodash/fp'
import { ALL_COMMUNITIES_ID } from 'components/Feed/Feed.store'

export default class AllCommunitiesFeed extends Component {
  componentDidMount () {
    this.props.fetchTopic()
  }

  componentDidUpdate (prevProps) {
    const { topicName, fetchTopic } = this.props
    if (topicName !== get('topicName', prevProps)) {
      fetchTopic()
    }
  }

  render () {
    const { currentUser, topic, topicName } = this.props
    const feedProps = {
      subject: 'all-communities',
      id: ALL_COMMUNITIES_ID,
      topic: get('id', topic),
      showCommunities: true,
      ...pick([
        'filter',
        'sortBy',
        'changeSort',
        'changeTab',
        'showPostDetails',
        'selectedPostId'
      ], this.props)
    }

    return <div styleName='container'>
      {!topicName && <FeedBanner all currentUser={currentUser} />}
      {topicName && <TopicFeedHeader
        postsTotal={get('postsTotal', topic)}
        followersTotal={get('followersTotal', topic)}
        topic={topic}
        topicName={topicName} />}
      <Feed {...feedProps} />
    </div>
  }
}
