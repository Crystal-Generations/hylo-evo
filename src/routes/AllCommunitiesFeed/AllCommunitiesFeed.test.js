import AllCommunitiesFeed from './AllCommunitiesFeed'
import { shallow } from 'enzyme'
import React from 'react'

it('renders as expected', () => {
  const wrapper = shallow(<AllCommunitiesFeed filter='request' />)
  expect(wrapper).toMatchSnapshot()
})

it('displays the regular FeedBanner if on the main feed', () => {
  const props = {}
  const wrapper = shallow(<AllCommunitiesFeed {...props} />)
  expect(wrapper.find('FeedBanner')).toHaveLength(1)
  expect(wrapper.find('TopicFeedHeader')).toHaveLength(0)
})

it('displays the TopicFeedHeader if on a topic feed', () => {
  const props = {
    topicName: 'petitions'
  }
  const wrapper = shallow(<AllCommunitiesFeed {...props} />)
  expect(wrapper.find('FeedBanner')).toHaveLength(0)
  expect(wrapper.find('Connect(TopicFeedHeader)')).toHaveLength(1)
})
