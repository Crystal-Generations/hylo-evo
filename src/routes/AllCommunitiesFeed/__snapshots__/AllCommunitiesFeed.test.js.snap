// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <FeedBanner
    all={true}
  />
  <Connect(Feed)
    filter="request"
    id="all-communities"
    subject="all-communities"
  />
</div>
`;
